docs/design_notes/lift.md

```diff
@@ -0,0 +1,355 @@
+# Lifted Transformations
+
+⚠️ Advanced topic ⚠️
+
+This design note explains the underlying implementation of `flax.linen.transform`, which enables JAX transformations inside `Module`s.
+
+
+## Introduction
+
+JAX uses a functional API meaning that it only guarantees correct behavior when using functions without side effects.
```

Comment:
You may want to cite https://jax.readthedocs.io/en/latest/jax-101/01-jax-basics.html#differences-from-numpy